research/2022-01-17-speeding-up-evm-part-1.mdx

```diff
@@ -0,0 +1,156 @@
+
+To improve EVM performance with backward compatibility and minimal change to consensus rules or how storage is implemented, we need parallelism.[^2]
+
+[^1]: Even though at the moment the biggest bottleneck of Ethereum is not EVM's performance (the size of the Ethereum state storage is), it will soon be as Ethereum TPS goes up.
+[^2]: Of course, many other optimizations don't need base layer changes, such as batching state read & write operations, caching, flattening the storage database, or just using a more performant programming language. We see many of those optimizations implemented in clients like [Akula](https://medium.com/@vorot93/meet-akula-the-fastest-ethereum-implementation-ever-built-58eaca244c39) and [Silkworm](https://medium.com/@giulio.rebuffo/silkworm-and-akula-the-future-of-erigon-fda4d6813505).
+
+__In part 1 of the series, we argue the need for a parallel EVM and cover general approaches to achieve it such as [EIP 648](https://ethresear.ch/t/parallelilizing-evm-through-end-of-the-block-virtual-transactions/7787) and [speculative execution](https://drops.dagstuhl.de/opus/volltexte/2020/11968/pdf/OASIcs-Tokenomics-2019-4.pdf).__ In part 2 we study how static analysis and formal methods can make the EVM parallelizable, specifically we present two simple algorithms for achieving parallelization.
+
+![](https://i.imgur.com/IZlQIFg.png)
+
+
+## Background
+
+Decentralization is vital to blockchain security so that it's hard for participants in the network to collude. [Ideally](https://vitalik.ca/general/2021/05/23/scaling.html), this is achieved by as many individual parties running nodes to verify ongoing transactions as possible. However it usually takes more than 3 days for regular users with a personal computer to spin up a full node on Ethereum (it requires days to re-compute and validate the 13 million blocks out there). Behind this inefficiency is [Ethereum's large storage size](https://vitalik.ca/general/2021/05/23/scaling.html) and more specifically [EVM's storage design](https://www.youtube.com/watch?v=2_GX8iCVNrA):
```

Comment:
_Decentralization is vital to blockchain security so that it's hard for participants in the network to collude._
This feels kinda reductive to what decentralization is for